desc/objectives/_profiles.py

```diff
@@ -0,0 +1,544 @@
+            Equilibrium that will be optimized to satisfy the Objective.
+        use_jit : bool, optional
+            Whether to just-in-time compile the objective and derivatives.
+        verbose : int, optional
+            Level of output.
+
+        """
+        eq = eq or self._eq
+        if self._grid is None:
+            grid = LinearGrid(
+                L=eq.L_grid,
+                M=eq.M_grid,
+                N=eq.N_grid,
+                NFP=eq.NFP,
+                sym=eq.sym,
+                axis=False,
+            )
+        else:
+            grid = self._grid
+
+        if isinstance(self._target, Profile):
+            self._target = self._target(grid.nodes[grid.unique_rho_idx])
+
+        self._dim_f = grid.num_rho
+        self._data_keys = ["rho", "iota", "iota_r"]
+        self._args = get_params(
+            self._data_keys,
+            obj="desc.equilibrium.equilibrium.Equilibrium",
+            has_axis=grid.axis.size,
+        )
+
+        timer = Timer()
+        if verbose > 0:
+            print("Precomputing transforms")
+        timer.start("Precomputing transforms")
+
+        self._profiles = get_profiles(self._data_keys, obj=eq, grid=grid)
+        self._transforms = get_transforms(self._data_keys, obj=eq, grid=grid)
+        self._constants = {
+            "transforms": self._transforms,
+            "profiles": self._profiles,
+        }
+
+        timer.stop("Precomputing transforms")
+        if verbose > 1:
+            timer.disp("Precomputing transforms")
+
+        super().build(eq=eq, use_jit=use_jit, verbose=verbose)
+
+    def compute(self, *args, **kwargs):
+        """Compute shear profile errors.
+
+        Parameters
+        ----------
+        R_lmn : ndarray
+            Spectral coefficients of R(rho,theta,zeta) -- flux surface R coordinate (m).
+        Z_lmn : ndarray
+            Spectral coefficients of Z(rho,theta,zeta) -- flux surface Z coordinate (m).
+        L_lmn : ndarray
+            Spectral coefficients of lambda(rho,theta,zeta) -- poloidal stream function.
+        i_l : ndarray
+            Spectral coefficients of iota(rho) -- rotational transform profile.
+        c_l : ndarray
+            Spectral coefficients of I(rho) -- toroidal current profile.
+        Psi : float
+            Total toroidal magnetic flux within the last closed flux surface (Wb).
+
+        Returns
+        -------
+        shear : ndarray
+            Normalized radial derivative of the rotational transform.
+
+        """
+        params, constants = self._parse_args(*args, **kwargs)
+        if constants is None:
+            constants = self.constants
+        data = compute_fun(
+            "desc.equilibrium.equilibrium.Equilibrium",
+            self._data_keys,
+            params=params,
+            transforms=constants["transforms"],
+            profiles=constants["profiles"],
+        )
+        shear = jnp.where(
+            data["iota"] == 0,
+            data["iota_r"],
```

Comment:
is this the correct limit?

Reply:
No. Doing l'Hospital's rule gives: $\frac{\rho\iota'}{\iota} \rightarrow 1 + \rho\frac{\iota''}{\iota'}$. 

But the case I'm really trying to handle is the default vacuum tokamak `Equilibrium()` that has zero rotational transform everywhere, and therefore all of the iota derivates will also be 0. So I'm not sure how to handle that case and I'm open to suggestions.

Reply:
hmm i see. Another possible tricky bit is that iota may not be exactly zero, but down to machine precision, so maybe we should have a threshold?

As for the limit, I think it would be good to implement the "correct" limit for cases that are more like a FRC where iota passes through zero.

For the case where iota is identically zero we could add something like 
`jnp.where(jnp.all(jnp.abs(iota)) < eps, 0, shear)`, to distinguish between iota being zero at a single point vs the whole profile

Reply:
Done, see the latest changes.

Reply:
The first call to jnp.where is only correct near $\rho = 0$. To do what you want, you would need something like
```python
indeterminate_1 = (jnp.abs(data["iota"]) < self._eps) & (jnp.abs(data["rho"]) < self._eps`
indeterminate_2 = (jnp.abs(data["iota"]) < self._eps) & (jnp.abs(data["iota_r"]) < self._eps`
shear = jnp.where(indeterminate_1 | indeterminate_2, limit, shear)
# The limit is undefined in this case so just set to zero.
shear = jnp.where(jnp.abs(iota).all() < eps, 0, shear)
```

Reply:
Does `jnp.nan` mess up the objective?

Reply:
Basically the requirement to apply lhopitals rule is:
```python
# iota needs to be close to zero, and then either rho is close to 0 or iota_r is close to 0
(jnp.abs(data["iota"]) < self._eps) & ((jnp.abs(data["rho"]) < self._eps) | (jnp.abs(data["iota_r"]) < self._eps))
```